custom/src/main/java/com/splunk/opentelemetry/jaeger/JaegerThriftSpanExporterFactory.java

```diff
@@ -16,23 +16,40 @@
 
 package com.splunk.opentelemetry.jaeger;
 
+import io.jaegertracing.thrift.internal.senders.HttpSender;
 import io.opentelemetry.exporter.jaeger.thrift.JaegerThriftSpanExporter;
 import io.opentelemetry.javaagent.spi.exporter.SpanExporterFactory;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import java.util.Collections;
 import java.util.Properties;
 import java.util.Set;
+import okhttp3.OkHttpClient;
 
-// TODO remove when https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1609
-// is implemented
 public class JaegerThriftSpanExporterFactory implements SpanExporterFactory {
+  static final String SIGNALFX_AUTH_TOKEN = "signalfx.auth.token";
+  static final String OTEL_EXPORTER_JAEGER_ENDPOINT = "otel.exporter.jaeger.endpoint";
+
   @Override
   public SpanExporter fromConfig(Properties config) {
-    return JaegerThriftSpanExporter.builder().readProperties(config).build();
+    JaegerThriftSpanExporter.Builder builder =
+        JaegerThriftSpanExporter.builder().readProperties(config);
+
+    String token = config.getProperty(SIGNALFX_AUTH_TOKEN, "");
+    if (!token.isEmpty()) {
+      builder.setThriftSender(
+          new HttpSender.Builder(config.getProperty(OTEL_EXPORTER_JAEGER_ENDPOINT))
+              .withClient(
+                  new OkHttpClient.Builder()
+                      .addInterceptor(new AuthTokenInterceptor(token))
+                      .build())
+              .build());
+    }
+
+    return builder.build();
   }
 
   @Override
   public Set<String> getNames() {
-    return Collections.singleton("jaeger-thrift");
+    return Collections.singleton("jaeger-thrift-splunk");
```

Comment:
I think we need a separate name since we're adding our custom logic - and the TODO comment is no longer viable here.

Reply:
TODO comment is indeed no longer applicable. But I am not sure if we need separate exporter name. If `SIGNALFX_AUTH_TOKEN` is not given, it will behave exactly like Otel's `jaeger-thrift`. I think it will be less confusing for clients to keep the same name. Assuming that we will not conflict with Otel when https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1609 is done

Reply:
> Assuming that we will not conflict with Otel when open-telemetry/opentelemetry-java-instrumentation#1609 is done

Unfortunately I think that we will have a conflict here. Otel javaagent takes the first exporter with a given name, so if we have 2 named jaeger-thrift then it's pretty much random which one gets chosen. Having a separate name for ours is more error-proof.

```java
  private static SpanExporterFactory findSpanExporterFactory(String exporterName) {
    ServiceLoader<SpanExporterFactory> serviceLoader =
        ServiceLoader.load(SpanExporterFactory.class, TracerInstaller.class.getClassLoader());

    for (SpanExporterFactory spanExporterFactory : serviceLoader) {
      if (spanExporterFactory.getNames().contains(exporterName)) {
        return spanExporterFactory;
      }
    }
    return null;
  }
```

Reply:
I agree. Now there is no difference between these two if TOKEN is empty, but the other may change in time so "shadowing" it could lead to some issues in the future.